local_planner/src/nodes/local_planner_node.cpp

```diff
@@ -193,18 +193,25 @@ void LocalPlannerNode::updatePlannerInfo() {
   // update the point cloud
   local_planner_->complete_cloud_.clear();
   for (size_t i = 0; i < cameras_.size(); ++i) {
-    sensor_msgs::PointCloud2 pc2cloud_world;
-    pcl::PointCloud<pcl::PointXYZ> complete_cloud;
+    pcl::PointCloud<pcl::PointXYZ> pcl_cloud_world_frame;
+    pcl::PointCloud<pcl::PointXYZ> pcl_cloud_cam_frame;
+    pcl::PointCloud<pcl::PointXYZ> pcl_cloud_cam_frame_without_nan;
     try {
-      tf::StampedTransform transform;
-      tf_listener_.lookupTransform(
-          "/local_origin", cameras_[i].newest_cloud_msg_.header.frame_id,
-          ros::Time(0), transform);
-      pcl_ros::transformPointCloud("/local_origin", transform,
-                                   cameras_[i].newest_cloud_msg_,
-                                   pc2cloud_world);
-      pcl::fromROSMsg(pc2cloud_world, complete_cloud);
-      local_planner_->complete_cloud_.push_back(std::move(complete_cloud));
+      // transform message to pcl type
+      pcl::fromROSMsg(cameras_[i].newest_cloud_msg_, pcl_cloud_cam_frame);
+
+      // remove nan padding
+      std::vector<int> dummy_index;
+      pcl::removeNaNFromPointCloud(
```

Comment:
According to http://docs.pointclouds.org/trunk/group__filters.html#gac463283a9e9c18a66d3d29b28a575064
The input and output cloud can be the same.

Also reserve dummy_index with the size of the input cloud, to prevent extra allocations.